### PR TITLE
[WIP] SE-0107: UnsafePointer and UnsafeRawPointer API changes.

### PR DESCRIPTION
**Pull request by atrick**
I'm including these changes related to SE-0107 in a PR in case anyone wants to add line-comment to the details doc comments or implementation.

### CODE REVIEW

File: stdlib/public/core/UnsafeRawPointer.swift.gyb [hunk 1]
```diff
@@ -0,0 +1,557 @@
...
+  /// Construct ${a_Self} with a given pattern of bits.
+  @_transparent
+  public init?(bitPattern: UInt) {
+    if bitPattern == 0 { return nil }
+    _rawValue = Builtin.inttoptr_Word(bitPattern._builtinWordValue)
+  }
+
+  /// Convert an Unsafe${Mutable}Pointer ${a_Self}.
+  @_transparent
+  public init<T>(_ other : Unsafe${Mutable}Pointer<T>) {
```
Comment (gribozavr) [inline stdlib/public/core/UnsafeRawPointer.swift.gyb]: The standard library style does not use spaces before colons in variable declarations.

File: test/1_stdlib/UnsafeRawPointer.swift [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    (ptr2 + 1).initialize(to: 1)
+    _ = f(UnsafeMutableRawPointer(ptr2 + 2))(as: Int.self, from: ptr2, count: 2)
+    expectEqual(0, ptr2[2])
+    expectEqual(1, ptr2[3])
+  }
+}
+
+func checkPtr(
+  _ f: ((UnsafeMutableRawPointer) -> (as: Int.Type, from: UnsafeMutablePointer<Int>, count: Int) -> UnsafeMutablePointer<Int>))
+-> (Check) -> Void {
```
Comment (gribozavr) [inline test/1_stdlib/UnsafeRawPointer.swift]: Please wrap function signatures like we do elsewhere:

```swift
func nameNameName(
  label name: TypeTypeType,
  label name: TypeTypeType,
  label name: TypeTypeType
) -> ReturnType {
}
```

File: stdlib/public/core/UnsafeRawPointer.swift.gyb [hunk 3]
```diff
@@ -0,0 +1,557 @@
...
+%  Mutable = 'Mutable' if mutable else ''
+
+/// A raw pointer for accessing untyped data. This provides no
+/// automatic memory management, no type safety, and no alignment
+/// guarantees. This implements Strideable to provide a view
+/// of byte-addressable memory.
+@_fixed_layout
+public struct Unsafe${Mutable}RawPointer : Strideable, Hashable, _Pointer {
+  /// The underlying raw pointer.
+  public let _rawValue: Builtin.RawPointer
```
Comment (gribozavr) [inline stdlib/public/core/UnsafeRawPointer.swift.gyb]: Why public?  Can we make it internal?

Comment (atrick) [inline stdlib/public/core/UnsafeRawPointer.swift.gyb]: I'm happy to make _rawValue internal. I have no idea what will break if we also make UnsafePointer._rawValue internal. There's no comment explaining why it was made public.

**Pull request closed without merging**